ssl/record/ssl3_record_tls13.c

```diff
@@ -53,21 +53,27 @@ int tls13_enc(SSL *s, SSL3_RECORD *recs, size_t n_recs, int send)
     }
     ivlen = EVP_CIPHER_CTX_iv_length(ctx);
 
+    if (EVP_CIPHER_CTX_mode(ctx) == EVP_CIPH_CCM_MODE) {
+        if (s->s3->tmp.new_cipher->algorithm_enc
+                & (SSL_AES128CCM8 | SSL_AES256CCM8))
+            taglen = EVP_CCM8_TLS_TAG_LEN;
+         else
+            taglen = EVP_CCM_TLS_TAG_LEN;
+         if (send && EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_SET_TAG, taglen,
+                                         NULL) <= 0)
+            return -1;
+    } else {
+        taglen = EVP_GCM_TLS_TAG_LEN;
```

Comment:
I kind of think we should have a branch here for chacha/poly...even it happens to be the same value. Failing that we should at least have a comment about it.